### ISSUE TITLE
Error while hydrating nextjs-ssr-delegate-modules example

### ISSUE THREAD

**Issue by kantesilvestrs** [module-federation/module-federation-examples#3041]
I ran locally the `nextjs-ssr-delegate-modules` example and it seems that no matter what I do the only way to prevent the hydration issue is by setting `ssr: false` - but it defeats the purpose of the example. 

Here is a quick vid showing the issue:

https://user-images.githubusercontent.com/52054299/235805106-a0b0a319-06da-4426-a3af-3553dc6947a8.mp4

**Comment by anhnguyentrong-ct** [comment]
any updates on this one ? i'm facing the same issues

**Comment by kubmir** [comment]
I have the same issue within my project. I have simple host app which includes remote - exposed component from different MF (renders only `<p>Test</p>`). It is rendered correctly on the server (I can see it in returned HTML). However, initial client render does not include it in HTML which causes hydration error. I import component like:

`const Header = dynamic(
  () => import('header/Header').then((mod) => mod.Header),
  {
    ssr: true,
  },
)`

Switching SSR to false make it "work" but only CSR. I tried it also with v7 but it is the same as with v6 of module federation package.

In other words, If I render `<><Header /><p>Test 2</p></> ` I get expected HTML on the server but `<p>Test 2</p>` on the client. That causes hydration error and fallback to CSR.

**Comment by kantesilvestrs** [comment]
@kubmir  I have noticed that `React.lazy` solves this issue for me - instead of using `next/dynamic`

**Comment by kubmir** [comment]
Thank you @kantesilvestrs ... I have discovered comment about version 7 and already implemented it. It really solves the issue.

**Comment by kantesilvestrs** [comment]
I solved this issue by using `React.lazy`
